clash-ghc/src-ghc/Clash/GHC/Evaluator.hs

```diff
@@ -1,4405 +1,445 @@
-  "Clash.Sized.Internal.BitVector.low"
-    -> reduce (mkBitLit ty 0 0)
-
-  "Clash.Sized.Internal.BitVector.undefined#"
-    | Just (_, kn) <- extractKnownNat tcm tys
-    -> let resTyInfo = extractTySizeInfo tcm ty tys
-           mask = bit (fromInteger kn) - 1
-       in reduce (mkBitVectorLit' resTyInfo mask 0)
-
--- Eq
-  "Clash.Sized.Internal.BitVector.eq##" | [(0,i),(0,j)] <- bitLiterals args
-    -> reduce (boolToBoolLiteral tcm ty (i == j))
-  "Clash.Sized.Internal.BitVector.neq##" | [(0,i),(0,j)] <- bitLiterals args
-    -> reduce (boolToBoolLiteral tcm ty (i /= j))
-
--- Ord
-  "Clash.Sized.Internal.BitVector.lt##" | [(0,i),(0,j)] <- bitLiterals args
-    -> reduce (boolToBoolLiteral tcm ty (i <  j))
-  "Clash.Sized.Internal.BitVector.ge##" | [(0,i),(0,j)] <- bitLiterals args
-    -> reduce (boolToBoolLiteral tcm ty (i >= j))
-  "Clash.Sized.Internal.BitVector.gt##" | [(0,i),(0,j)] <- bitLiterals args
-    -> reduce (boolToBoolLiteral tcm ty (i >  j))
-  "Clash.Sized.Internal.BitVector.le##" | [(0,i),(0,j)] <- bitLiterals args
-    -> reduce (boolToBoolLiteral tcm ty (i <= j))
-
--- Bits
-  "Clash.Sized.Internal.BitVector.and##"
-    | [i,j] <- bitLiterals args
-    -> let Bit msk val = BitVector.and## (toBit i) (toBit j)
-       in reduce (mkBitLit ty (toInteger msk) (toInteger val))
-  "Clash.Sized.Internal.BitVector.or##"
-    | [i,j] <- bitLiterals args
-    -> let Bit msk val = BitVector.or## (toBit i) (toBit j)
-       in reduce (mkBitLit ty (toInteger msk) (toInteger val))
-  "Clash.Sized.Internal.BitVector.xor##"
-    | [i,j] <- bitLiterals args
-    -> let Bit msk val = BitVector.xor## (toBit i) (toBit j)
-       in reduce (mkBitLit ty (toInteger msk) (toInteger val))
-
-  "Clash.Sized.Internal.BitVector.complement##"
-    | [i] <- bitLiterals args
-    -> let Bit msk val = BitVector.complement## (toBit i)
-       in reduce (mkBitLit ty (toInteger msk) (toInteger val))
-
--- Pack
-  "Clash.Sized.Internal.BitVector.pack#"
-    | [(msk,i)] <- bitLiterals args
-    -> let resTyInfo = extractTySizeInfo tcm ty tys
-       in  reduce (mkBitVectorLit' resTyInfo msk i)
-
-  "Clash.Sized.Internal.BitVector.unpack#"
-    | [(msk,i)] <- bitVectorLiterals' args
-    -> reduce (mkBitLit ty msk i)
-
--- Concatenation
-  "Clash.Sized.Internal.BitVector.++#" -- :: KnownNat m => BitVector n -> BitVector m -> BitVector (n + m)
-    | Just (_,m) <- extractKnownNat tcm tys
-    , [(mski,i),(mskj,j)] <- bitVectorLiterals' args
-    -> let val = i `shiftL` fromInteger m .|. j
-           msk = mski `shiftL` fromInteger m .|. mskj
-           resTyInfo = extractTySizeInfo tcm ty tys
-       in reduce (mkBitVectorLit' resTyInfo msk val)
-
--- Reduction
-  "Clash.Sized.Internal.BitVector.reduceAnd#" -- :: KnownNat n => BitVector n -> Bit
-    | [i] <- bitVectorLiterals' args
-    , Just (_, kn) <- extractKnownNat tcm tys
-    -> let resTy = getResultTy tcm ty tys
-           val = reifyNat kn (op (toBV i))
-       in reduce (mkBitLit 
```

Comment:
I know this is copied from the old source location, but reviewing it now I see that we're missing a case for `Clash.Sized.Internal.BitVector.fromInteger##` : the "constructor" for `Bit`.

Reply:
Ugh, github messed up the reference to the lines I was talking about in this review comment. I'm was talking about this expression: https://github.com/clash-lang/clash-compiler/blob/8f6a8a7a2ff5ad84a07351360fc9736ba67370ed/clash-ghc/src-ghc/Clash/GHC/Evaluator.hs#L386-L395

Reply:
@christiaanb, just as a sanity check, `fromInteger##` should only have `[Lit l0] <- vs`, as the first argument on the others should be the `KnownNat`?

Reply:
No, it's `[Lit (WordLiteral 0), Lit l0] <- vs` because just like `BitVector`, `Bit` also has a mask bit. 

Also, I dunno how I missed this when I made the change to `Clash.Sized.Internal.BitVector.fromInteger#`, but we also have to change the pattern for that in the case-expression to: `[_,Lit (NaturalLiteral 0),Lit l0] <- vs`